packages/react-router/src/link.tsx

```diff
@@ -84,67 +84,78 @@ export type RemoveLeadingSlashes<T> = T extends `/${infer R}`
   ? RemoveLeadingSlashes<R>
   : T
 
+export type ResolvePaths<TRouteTree extends AnyRoute, TSearchPath> =
```

Comment:
For relative routes starting with '../' and '.' we can narrow the union of the paths down to the children of the resolved path. 

Basically, because we know `TFrom` is a route. We can actually only get descendants of the resolved route